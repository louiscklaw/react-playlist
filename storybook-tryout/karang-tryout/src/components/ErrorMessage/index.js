import React from 'react';
import { node } from 'prop-types';
import styled, { css } from 'styled-components';

import { valencia } from 'styles/colors';
import { primaryFonts, fontSize } from 'styles/fonts';
import WarningIcon from 'components/Icon/icons/alert/warning';

const Container = styled.div`
  display: flex;
  align-items: center;
  margin-top: 0.3em;
  margin-bottom: 0.3em;
  line-height: 1.4;
  ${({ theme: { rtl } }) =>
    rtl &&
    css`
      direction: rtl;
    `};
`;

const Text = styled.span`
  flex-grow: 1;
  width: 0;
  color: ${valencia.main};
  font-family: ${primaryFonts};
  font-size: ${fontSize.small};
`;

const IconWrapper = styled.span`
  ${({ theme: { rtl } }) =>
    !rtl &&
    css`
      margin-left: 1em;
    `}
`;

const ErrorMessage = ({ error, ...rest }) =>
  error &&
  error.length > 0 && (
    <Container {...rest}>
      <Text>{error}</Text>
      <IconWrapper>
        <WarningIcon color={valencia.main} size={13} />
      </IconWrapper>
    </Container>
  );

ErrorMessage.propTypes = {
  error: node,
};

ErrorMessage.defaultProps = {
  error: null,
};

export default ErrorMessage;
