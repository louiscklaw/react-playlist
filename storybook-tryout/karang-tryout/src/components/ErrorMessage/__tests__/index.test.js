import React from 'react';
import { render } from 'enzyme';

import ErrorMessage from '../index';

describe('Snapshots', () => {
  it('ErrorMessage', () => {
    const wrapper = render(<ErrorMessage error="message">blah</ErrorMessage>);
    expect(wrapper).toMatchSnapshot();
  });
});
