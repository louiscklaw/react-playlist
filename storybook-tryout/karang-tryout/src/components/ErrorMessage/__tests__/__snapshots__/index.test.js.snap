// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots ErrorMessage 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0.3em;
  margin-bottom: 0.3em;
  line-height: 1.4;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 0;
  color: #D8534F;
  font-family: Noto Sans,"Noto Sans TC",sans-serif;
  font-size: 12px;
}

.c2 {
  margin-left: 1em;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    message
  </span>
  <span
    class="c2"
  >
    <svg
      aria-hidden="false"
      aria-labelledby="title"
      focusable="false"
      role="img"
      style="vertical-align:middle;fill:#D8534F;width:13px;height:13px"
      viewBox="0 0 217 217"
    >
      <title>
        Warning
      </title>
      <g>
        <rect
          fill="none"
          height="217"
          width="217"
        />
        <path
          d="M109,36.62a9.23,9.23,0,0,1,6.56,2.7,9.74,9.74,0,0,1,1.83,2.61,9.6,9.6,0,0,0,.65,1.12l70.43,122,.15.24a9.27,9.27,0,0,1-8,14.14l-.66,0c-.2,0-.41,0-.62,0h-.52l-141,0h-.56a9.24,9.24,0,0,1-8-4.62,9.29,9.29,0,0,1-.93-7.07l0-.05a8.74,8.74,0,0,1,1.26-2.73,8.13,8.13,0,0,0,.58-1L100.53,42c.12-.19.23-.38.34-.59a8.81,8.81,0,0,1,1.56-2.1A9.23,9.23,0,0,1,109,36.62M31.17,159.81h0M109,27.62A18.22,18.22,0,0,0,96.07,33a17.53,17.53,0,0,0-3.17,4.28h0L22.17,159.81v0a17.73,17.73,0,0,0-2.54,5.52,18.24,18.24,0,0,0,17.65,23h.56l141.53,0h0c.43,0,.85,0,1.28,0a18.27,18.27,0,0,0,15.62-27.89L125.49,38h0a18.68,18.68,0,0,0-3.53-5A18.22,18.22,0,0,0,109,27.62Z"
        />
        <circle
          cx="109"
          cy="158.63"
          r="9"
        />
        <path
          d="M118,131.5a9,9,0,0,1-18,0V77.37a9,9,0,0,1,18,0Z"
        />
      </g>
    </svg>
  </span>
</div>
`;
